src/libYARP_OS/include/yarp/os/Network.h

```diff
@@ -592,6 +592,14 @@ class YARP_OS_API yarp::os::NetworkBase {
      */
     static bool getConnectionQos(const ConstString& src, const ConstString& dest,
                                  QosStyle& srcStyle, QosStyle& destStyle, bool quiet=true);
+    /**
+     * Checks that the port has a valid name. To be valid the portName
+     * should start with '/', not contain spaces, not be empty and not end with '/'.
```

Comment:
What you mean by "spaces" ? See https://en.wikipedia.org/wiki/Whitespace_character#Definition_and_ambiguity .

Reply:
I didn't know there are so many "spaces" :scream: I mean `' '` character

Reply:
Are we sure that we want to allow ports names that contain a tab? 
See https://github.com/ros2/design/pull/90#issuecomment-221784192 for a related discussion.

Reply:
See http://design.ros2.org/articles/topic_and_service_names.html for the updated article.

Reply:
I can add the check for '/t' character :wink:

Reply:
I have the feeling that if we don't deal with this systematically we will have problems, especially if we promise to support "strange" port names on which people will start to depend. 

@drdanz What is your take on this?

Reply:
I.e. can '<' be part of a valid YARP portname?

Reply:
I'm aligned with @drdanz point of view. I would just keep thing as they are, i.e. quite loose on port names, and open a discussion with seniors like @lornat75 to discuss about a name policy.
After that we can safely define what is _valid_ and implement the proper checks.

Reply:
Ok!

Reply:
Right now the only constraint are:
* a port must start with `/`
* a port must not contains `@`  , because _if present_, it is interpreted as a (ROS) topic
* `:` is allowed

Right now I think that  no special policy is enforced for handling other special character, therefore the result is a matter of implementation.

For example, right now, if you do:
`yarp write /#~=*  `   --> opens a port with name   `/#~=*`

But when writing on terminal, some special character are interpreted by bash itself
`yarp write /name&surname` for bash is equivalent to
```
yarp write /name &
surname
``` 
resulting in a port with name `/name` and 
` surname: command not found`

Using `"` will fix this problem
`yarp write "/name&surname"`  --> opens a port with name `/name&surname`

Same happens for `(` .
`$` is always interpreted, even inside `" `

Both
`yarp write /$YARP_DIR`  and  `yarp write "/$YARP_DIR"`  will open a port with name 
`//usr/local/src/robot/ROBOT_CODE/yarp/build-64`  

When used in code instead characters are never interpreted.
This means that using `$` will end in a different result when used from terminal or from code.

Reply:
Edit:
using escape `\` before `$` force it to not be interpreted. (I forgot this option)
`yarp write /\$YARP_DIR` will succesfully open a port with name ` /$YARP_DIR`

Anyway this may be shell-dependent. Is Win behaviour the same?
Shall we rely on / allow it?

Reply:
Actually `yarp read /a-@/b` creates 2 ports: `/b` and `/a-@/b`. You can then connect normally to the second port (for example `yarp write ... /a-@/b`) and this works normally...

Reply:
Exactly. Maybe I wasn't clear. I didn't meant that when the @ is present, the port does not work, or that character must not be used.
I mean that @ is a kind of special/escape character for names, because it makes the port behave differently from a 'normal' one.

If my intention is to open a normal port and I use the @ in the name, then seeing 2 ports opened is something I might not expect or want. In other words, the user must know in advance that using @ in the names has this side-effect.

So, if we want to define guidelines for port name, I'd suggest to limit the usage the @ only when dealing with ROS topic, and to avoid it when only yarp ports are involved.